src/services/userService.js

```diff
@@ -0,0 +1,51 @@
+let users = [];
+let userIdCounter = 1;
+
+const resetUsers = () => {
+  users = [];
+  userIdCounter = 1;
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: userIdCounter++,
+    name,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(+id);
+
+  if (!user) {
+    return null;
+  }
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const deleteUser = (id) => {
+  users = users.filter((person) => person.id !== +id);
+};
```

Comment:
The deleteUser function currently doesn't return anything. It might be useful to return a boolean indicating whether the deletion was successful or not. This way, the caller of the function can handle the case where the user to delete doesn't exist.